src/hooks/use-fetch-wpcom-sites.tsx

```diff
@@ -45,8 +51,19 @@ type SitesEndpointResponse = {
 
 const STUDIO_SYNC_FEATURE_NAME = 'studio-sync';
 
+function isJetpackSite( site: SitesEndpointSite ): boolean {
+	return !! site.jetpack && ! site.is_wpcom_atomic;
+}
+
+function hasSupportedPlan( site: SitesEndpointSite ): boolean {
+	return !! site.plan && site.plan.features.active.includes( STUDIO_SYNC_FEATURE_NAME );
+}
+
 function getSyncSupport( site: SitesEndpointSite, connectedSiteIds: number[] ): SyncSupport {
-	if ( ! site.plan || ! site.plan.features.active.includes( STUDIO_SYNC_FEATURE_NAME ) ) {
+	if ( isJetpackSite( site ) && ! hasSupportedPlan( site ) ) {
```

Comment:
It might be fine just to check for the Jetpack site but I added the extra check for the plan as well. Happy to remove it. I feel like there are so many intricacies regarding JP sites and it is better to be on the safer side